### ISSUE TITLE
adding additional puzzles resp. links to more puzzles

### ISSUE THREAD

**Issue by DJCrashdummy** [ogarcia/opensudoku#21]
beforehand: thank you sooo much for bringing this awesome project back to life!!!

i know, there is a generator at the [homepage](https://opensudoku.moire.org/) (kudos for that!)... but how about some additional links to static collections (like at the [original project](https://code.google.com/archive/p/opensudoku-android/)) for easily adding the same big collections on several devices?
beside the [original project's collection](https://code.google.com/archive/p/opensudoku-android/wikis/Puzzles.wiki), https://opensudoku.p43.eu/ is really worth mentioning, and also at http://www.sudocue.net/download.php some collections can be found.

may these also be added also to the readme and the store-descriptions...?

**Comment by ogarcia** [comment]
Hi @DJCrashdummy 

In the original project are same files that in http://opensudoku.moire.org/ (The Easy, Medium, Hard and Very Hard puzzles). But I can add a link to https://opensudoku.p43.eu/ in the list. In http://www.sudocue.net/download.php cannot find any `.opensudoku` file. I cannot add links that not have files that can be added easy to game. Take note that we have non tecnical users.

**Comment by ogarcia** [comment]
Added P43 link into http://opensudoku.moire.org/ page.

**Comment by DJCrashdummy** [comment]
@ogarcia thanks for adding https://opensudoku.p43.eu/ to (at least) the homepage, but i see no reason for the hint (_The first list of gnome-sudoku/opensudoku files_), because `.sdm`-files can be added in the same way as `.opensudoku`-files.

**Comment by ogarcia** [comment]
@DJCrashdummy the hint was incomplete :-D ¡Fixed!

**Comment by DJCrashdummy** [comment]
sorry for bothering you with this... but for me importing `sdm`-files work **exactly** the same way as `opensudoku`-files, so also by opening the file via a file manager with _Open Sudoku_ the file gets imported.
therefore i see no reason to complicate things (even not explanations) for non-techy users... :confused:

**Comment by ogarcia** [comment]
_.opensudoku_ files can be imported directly from browser window but if you try to open _.sdm_ files directly with Open Sudoku the process fail. You must download first the file. That is the reason of the hint.

**Comment by DJCrashdummy** [comment]
ok... now i got you! :wink: 

i promise my last suggestion concerning this topic... why not use a more precise wording (since for example both files have to be "downloaded" - and i also experience every now and then confusion at my noobs with eventual differences in opening/importing):
_The first list of gnome-sudoku/opensudoku files can be <s>openend</s> **imported** directly by **the** OpenSudoku app, the second one must be <s>downloaded</s> **saved to the device** and imported by <s>hand</s> **either opening it with the OpenSudoku app or** using the import option in the app menu._

**Comment by ogarcia** [comment]
Done!

**Comment by chrisawad** [comment]
Btw, the generated files on the website are .opensudoku.xml. We have to rename the file and remove .xml to get it to import.

Can you either add xml extension to the intent filters or fix the website to generate without xml extension? 

Also it would be awesome to add a generate button in game or import from online option that downloads from the link without leaving the game. I would love this feature as solving the 20 very hard puzzles for me is cake and I'm having to revisit the site a million times.

**Comment by ogarcia** [comment]
@chrisawad the generated files in website are `.opensudoku`. I don't know if your browser is adding the .xml extension. Please try with Chrome or Firefox.

**Comment by DJCrashdummy** [comment]
i can confirm, that downloading one of the generated files with firefox or chromium (no matter if desktop or android) gives you a `.opensudoku` file.

**Comment by chrisawad** [comment]
> @chrisawad the generated files in website are `.opensudoku`. I don't know if your browser is adding the .xml extension. Please try with Chrome or Firefox.

Interesting. I just tried it on Chrome for desktop and it didn't add .xml. Chrome for Android on the other hand, always adds .xml. Considering this is an Android app and the link in the app opens for most people in Chrome for Android, they'll all face what I have to face every time I go to download more generated puzzles (open in Files app, remove .xml, then go back to the app).

I took a look at the code -- it should be pretty simple to pass the URI for the very hard generate link directly to OpenSudokuImportTask to avoid having to visit the website, right?

As someone who plays 5-10 very hard puzzles a day and having to go through this .xml renaming process just to get 20 new puzzels, this gets very annoying. 

Do we know if the puzzle generation logic on the website can be incorporated directly into the app? This too would skyrocket the usefulness of the app, with unlimited puzzles.
